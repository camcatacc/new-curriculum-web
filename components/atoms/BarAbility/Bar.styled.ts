// Modules
import styled from "styled-components";
import { motion } from "framer-motion";

// Elements
import { Typography as MaterialTypography } from "@material-ui/core";

// Components
export const ParentContainer = styled(motion.div)<{ $percentage: number; $backgroundColor: string }>`
	width: ${({ $percentage }) => $percentage}%;
	background-color: ${({ $backgroundColor }) => $backgroundColor};
	height: "100%";
`;

export const Typography = styled(MaterialTypography)<{ fontSize: string | number; fontcolor: string }>`
	font-size: ${({ fontSize }) => fontSize};
	color: ${({ fontcolor }) => fontcolor};
`;
