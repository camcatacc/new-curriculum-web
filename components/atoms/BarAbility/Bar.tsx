// Modules
import React from "react";

// Elements
import { Typography } from "@material-ui/core";
import { motion } from "framer-motion";

// Definitions
import type { Variants } from "framer-motion";

export interface BarProps {
	percentage: number;
	backgroundColor: string;
	fontSize?: number | string;
	fontColor?: string;
	animated?: boolean;
}

// Auxiliary functions and options
const getMotionOptions = (animated: boolean, percentage: number) => {
	if (!animated) return {};
	const parentVariants: Variants = {
		offscreen: {
			width: 0
		},
		onscreen: {
			width: `${percentage}%`,
			transition: {
				type: "tween",
				duration: 0.5,
				when: "beforeChildren"
			}
		}
	};

	const childVariants: Variants = {
		offscreen: {
			opacity: 0
		},
		onscreen: {
			opacity: 1,
			transition: {
				type: "tween",
				duration: 0.3
			}
		}
	};
	return { parentVariants, childVariants };
};

// Element
const Bar = ({ percentage, backgroundColor, fontSize = "1rem", fontColor = "white", animated = false }: BarProps) => {
	const { parentVariants, childVariants } = getMotionOptions(animated, percentage);
	return (
		<div className="bg-gray-300 h-full">
			<motion.div
				variants={parentVariants}
				style={{
					width: `${percentage}%`,
					backgroundColor,
					height: "100%"
				}}
				className="pr-4 h-full flex flex-col justify-center"
			>
				<motion.div variants={childVariants}>
					<Typography variant="body1" align="right" style={{ fontSize: fontSize, color: fontColor }}>{`${percentage}%`}</Typography>
				</motion.div>
			</motion.div>
		</div>
	);
};

export default Bar;
