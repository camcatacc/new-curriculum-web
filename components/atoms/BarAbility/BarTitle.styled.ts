// Modules
import styled from "styled-components";

// Elements
import { Typography as MaterialTypography } from "@material-ui/core";

// Components
export const Container = styled.div<{ backgroundColor?: string }>`
	background-color: ${({ backgroundColor }) => backgroundColor};
`;

export const Typography = styled(MaterialTypography)<{ fontcolor: string; fontSize: number | string }>`
	color: ${({ fontcolor }) => fontcolor};
	font-size: ${({ fontSize }) => fontSize};
`;
