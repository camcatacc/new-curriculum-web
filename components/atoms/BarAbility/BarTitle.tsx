// Modules
import React from "react";

// Elements
import { Typography } from "@material-ui/core";

// Definitions
export interface BarTitleProps {
	backgroundColor: string;
	fontColor?: string;
	title: any;
	fontSize?: number | string;
}

// Element
const BarTitle = ({ backgroundColor, title, fontSize = "1rem", fontColor = "white" }: BarTitleProps) => {
	return (
		<div className="pl-4 pr-4 pt-2 pb-2" style={{ backgroundColor: backgroundColor }}>
			<Typography variant="body1" style={{ color: fontColor, fontSize: fontSize }} className="w-full truncate" align="center">
				{title}
			</Typography>
		</div>
	);
};

export default BarTitle;
