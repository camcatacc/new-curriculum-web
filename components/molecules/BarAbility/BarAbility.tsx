// Modules
import React from "react";

// Elements
import Bar from "components/atoms/BarAbility/Bar";
import BarTitle from "components/atoms/BarAbility/BarTitle";

// Definitions
export interface BarAbilityProps {
	percentage: number;
	name: string;
	titleColor: string;
	barColor: string;
	fontColor?: string;
}

// Element
const BarAbility = ({ percentage, name, titleColor, barColor, fontColor }: BarAbilityProps) => (
	<div className="flex flex-row justify-center">
		<div className="w-32">
			<BarTitle backgroundColor={titleColor} fontColor={fontColor} title={name} />
		</div>
		<div className="w-full md:w-80 ">
			<Bar backgroundColor={barColor} percentage={percentage} fontColor={fontColor} />
		</div>
	</div>
);

export default BarAbility;
