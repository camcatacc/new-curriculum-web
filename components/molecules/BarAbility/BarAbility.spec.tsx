// Modules
import React from "react";
import { render } from "@testing-library/react";

// Elements
import BarAbility from "components/molecules/BarAbility/BarAbility";

// Definitions
import type { BarAbilityProps } from "components/molecules/BarAbility/BarAbility";

// Default Props
const defaultProps: BarAbilityProps = {
	percentage: 80,
	barColor: "red",
	name: "Cool",
	titleColor: "white"
};

// Tests
describe("BarAbility", () => {
	it("Renders as expected", () => {
		const { container } = render(<BarAbility {...defaultProps} />);
		expect(container).toMatchSnapshot();
	});
});
