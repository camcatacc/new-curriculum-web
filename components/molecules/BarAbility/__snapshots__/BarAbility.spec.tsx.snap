// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarAbility Renders as expected 1`] = `
<div>
  <div
    class="flex flex-row justify-center"
  >
    <div
      class="w-40 md:w-32 truncate"
    >
      <div
        class="sc-dkPtRN kLozLK pl-2 pr-2 pt-1 pb-1"
      >
        <p
          class="MuiTypography-root sc-hKwDye dsVJSv w-full truncate MuiTypography-body2 MuiTypography-alignCenter"
          font-size="1rem"
          fontcolor="white"
        >
          Cool
        </p>
      </div>
    </div>
    <div
      class="w-full md:w-80 "
    >
      <div
        class="bg-gray-300 h-full"
      >
        <div
          class="sc-bdvvtL kOgXaY pr-4 h-full flex flex-col justify-center"
        >
          <div>
            <p
              class="MuiTypography-root sc-gsDKAQ etFjmY MuiTypography-body1 MuiTypography-alignRight"
              font-size="1rem"
              fontcolor="white"
            >
              80%
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
